components/brave_referrals/browser/brave_referrals_service.h

```diff
@@ -30,7 +30,8 @@ std::string GetAPIKey();
 
 class BraveReferralsService {
  public:
-  explicit BraveReferralsService(PrefService* pref_service);
+  explicit BraveReferralsService(PrefService* pref_service,
+      std::string platform, std::string api_key);
```

Comment:
`platform` and `api_key` should be const references.

Reply:
I made them `const` and force pushed. Both of those are on the stack in the caller (factory function). Both the values will be elided

Reply:
Ohh, I was just talking about the params to this function not the member variables. If you make them `const std::string&` in this function, then we won't pass the entire string on the stack. Not a big deal, but it's usually a good idea.

Reply:
https://github.com/brave/brave-core/blob/f979e0a9d688abc0bbfbf4fae02b36bf935d4795/browser/brave_referrals/brave_referrals_service_factory.cc#L19-L20

I mean both those values are on the stack so a reference won't be valid. But the params are a function call and those will both be copy-elided- it should be the same overhead as using a reference.